infer/src/base/ProcessPool.ml

```diff
@@ -191,15 +191,8 @@ let killall pool ~slot status =
 
 
 let has_dead_child pool =
-  let open Option.Monad_infix in
-  Unix.wait_nohang `Any
-  >>= fun (dead_pid, status) ->
-  (* Some joker can [exec] an infer binary from a process that already has children. When some of
-     these pre-existing children die they'll get detected here but won't appear in our list of
-     workers. Just return [None] in that case. *)
```

Comment:
What happens in that case now? It cannot happen on Windows but in theory could happen on Unix.

Reply:
I'm not sure I understand the question. For me the comment meant there was basically a problem with the argument `'Any` to `wait_nohang`. With the new code logic, we always `wait` for our own children, and nothing more.